mod-audit-server/src/main/java/org/folio/dao/acquisition/impl/InvoiceLineEventsDaoImpl.java

```diff
@@ -0,0 +1,115 @@
+import java.time.ZoneOffset;
+import java.util.Date;
+import java.util.UUID;
+
+import io.vertx.core.Future;
+import io.vertx.core.json.JsonObject;
+import io.vertx.sqlclient.Row;
+import io.vertx.sqlclient.RowSet;
+import io.vertx.sqlclient.Tuple;
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.folio.dao.acquisition.InvoiceLineEventsDao;
+import org.folio.rest.jaxrs.model.InvoiceLineAuditEvent;
+import org.folio.rest.jaxrs.model.InvoiceLineAuditEventCollection;
+import org.folio.util.PostgresClientFactory;
+import org.springframework.stereotype.Repository;
+
+@Repository
+public class InvoiceLineEventsDaoImpl implements InvoiceLineEventsDao {
+
+  private static final Logger LOGGER = LogManager.getLogger();
+
+  public static final String TABLE_NAME = "acquisition_invoice_line_log";
+
+  public static final String GET_BY_INVOICE_LINE_ID_SQL = "SELECT id, action, invoice_id, invoice_line_id, user_id, event_date, action_date, modified_content_snapshot," +
+    " (SELECT count(*) AS total_records FROM %s WHERE invoice_line_id = $1) " +
+    " FROM %s WHERE invoice_line_id = $1 %s LIMIT $2 OFFSET $3";
+
+  private static final String INSERT_SQL = "INSERT INTO %s (id, action, invoice_id, invoice_line_id, user_id, event_date, action_date, modified_content_snapshot) " +
+    "VALUES ($1, $2, $3, $4, $5, $6, $7, $8)";
+
+  private final PostgresClientFactory pgClientFactory;
+
+  public InvoiceLineEventsDaoImpl(PostgresClientFactory pgClientFactory) {
+    this.pgClientFactory = pgClientFactory;
+  }
+
+  @Override
+  public Future<RowSet<Row>> save(InvoiceLineAuditEvent invoiceLineAuditEvent, String tenantId) {
+    LOGGER.debug("save:: Saving InvoiceLine AuditEvent with tenant id : {}", tenantId);
+    String logTable = formatDBTableName(tenantId, TABLE_NAME);
+    String query = format(INSERT_SQL, logTable);
+    return makeSaveCall(query, invoiceLineAuditEvent, tenantId)
+      .onSuccess(rows -> LOGGER.info("save:: Saved InvoiceLine AuditEvent with tenant id : {}", tenantId))
+      .onFailure(e -> LOGGER.error("Failed to save record with id: {} for invoice line id: {} in to table {}",
+        invoiceLineAuditEvent.getId(), invoiceLineAuditEvent.getInvoiceLineId(), TABLE_NAME, e));
+  }
+
+  @Override
+  public Future<InvoiceLineAuditEventCollection> getAuditEventsByInvoiceLineId(String invoiceLineId, String sortBy, String sortOrder, int limit, int offset, String tenantId) {
+    LOGGER.debug("getAuditEventsByInvoiceLineId:: Retrieving AuditEvent with invoice line id : {}", invoiceLineId);
+    String logTable = formatDBTableName(tenantId, TABLE_NAME);
+    String query = format(GET_BY_INVOICE_LINE_ID_SQL, logTable, logTable, format(ORDER_BY_PATTERN, sortBy, sortOrder));
+    return pgClientFactory.createInstance(tenantId).execute(query, Tuple.of(UUID.fromString(invoiceLineId), limit, offset))
+      .map(rowSet -> rowSet.rowCount() == 0 ? new InvoiceLineAuditEventCollection().withTotalItems(0)
```

Comment:
Better to put the condition results on separate lines for readability

Reply:
> Better to put the condition results on separate lines for readability

A better option is to extract into a method, can be done in later PRs for all classes